### PR TITLE
Added config.yaml support + thresholds overrides

### PR DESCRIPTION
**Pull request by mateuszmidor**
KNTK-221

### CODE REVIEW

File: presentation/main_view.py
Comment (mmac-m3a) [inline presentation/main_view.py:9]: (nit) If you put `from config import Config` in `domain.config.__init__.py`, you won't need the funny looking `domain.config.config` path in the import statement.
It is IMHO a good practice to put "public" importable objects in `__init__.py` in each module.

File: presentation/matrix_view.py
Comment (mmac-m3a) [inline presentation/matrix_view.py:56]: Ah, why not in this one? It is kind of unpleasant to have all of the configuration mechanics, but not the code using the config.

**Review by mmac-m3a — APPROVED** [review]
OK to go in.

**Comment by mmac-m3a** [comment]
The PR is fine with me, but I'll wait for approval from a Codilime team member.

File: infrastructure/config/thresholds.py
Comment (mateuszmidor) [reply infrastructure/config/thresholds.py:61]: will add general exception logging on top level (main.py)

File: presentation/matrix_view.py
Comment (mateuszmidor) [reply presentation/matrix_view.py:56]: OK, adding to current PR

**Review by Fenthick — CHANGES_REQUESTED** [review]

File: config.yaml
Comment (Fenthick) [reply config.yaml]: Done

File: infrastructure/config/config_yaml.py
Comment (mmac-m3a) [inline infrastructure/config/config_yaml.py]: We can really use `SafeLoader` (`config = yaml.safe_load(file)`), because we will never have to construct any complex objects or use references and overrides.

**Review by mmac-m3a — CHANGES_REQUESTED** [review]
Please, change YAML loading to use SafeLoader.

File: domain/config/thresholds.py
Comment (mateuszmidor) [reply domain/config/thresholds.py:4]: Separation of concerns; Thresholds can be imported and used without Config

File: infrastructure/config/config_yaml.py
Comment (mateuszmidor) [reply infrastructure/config/config_yaml.py:7]: Class name already says it provides Config from YAML.

File: infrastructure/config/config_yaml.py
Comment (mateuszmidor) [reply infrastructure/config/config_yaml.py:36]: good point; SafeLoader will suffice

File: infrastructure/data_access/http/synthetics_repo.py
Comment (mateuszmidor) [inline infrastructure/data_access/http/synthetics_repo.py:44]: NOTE: this is removed in next PR

File: domain/config/thresholds.py
Comment (Fenthick) [reply domain/config/thresholds.py:4]: If Thresholds is put to config.py module, it can still be imported independently with `from domain.config.config import Thresholds` (or `from domain.config import Thresholds` with `domain.config.__init__.py` improvement). And doing one tiny class per file IMO breaks KISS principle.

File: infrastructure/config/config_yaml.py
Comment (Fenthick) [reply infrastructure/config/config_yaml.py:7]: I prefer class comments to describe what the class does instead of how it is used or what interfaces it implements.

File: infrastructure/config/config_yaml.py
Comment (Fenthick) [reply infrastructure/config/config_yaml.py]: Done

File: domain/config/thresholds.py
Comment (Fenthick) [reply domain/config/thresholds.py:4]: I see that you merged Thresholds and ThresholdOverride. It is sensible now.

File: infrastructure/config/thresholds.py
Comment (Fenthick) [reply infrastructure/config/thresholds.py]: Done

**Review by Fenthick — APPROVED** [review]

**Review by lukaa12 — APPROVED** [review]

**Review by mmac-m3a — APPROVED** [review]

**Pull request merged**
